### DISCUSSION TITLE
`(void-function internal--without-restriction)` Under Emacs 30

### DISCUSSION THREAD

**Discussion by adham-omran** [minad/consult#1042] [Q&A] [answered]
I've started using Emacs 30 under NixOS lately, using the [nix-community/emacs-overlay](https://github.com/nix-community/emacs-overlay), I encountered this error when calling `(consult-imenu)`

```
Debugger entered--Lisp error: (void-function internal--without-restriction)
  internal--without-restriction(#f(compiled-function () #<bytecode 0x29d15cab85311>))
  consult-imenu--compute()
  consult-imenu--items()
  #f(compiled-function () #<bytecode 0x1b2a2ea3870d67bc>)()
  funcall-with-delayed-message(1 "Building Imenu..." #f(compiled-function () #<bytecode 0x1b2a2ea3870d67bc>))
  consult-imenu()
  (progn (consult-imenu))
  eval((progn (consult-imenu)) t)
  elisp--eval-last-sexp(nil)
  #f(compiled-function () #<bytecode 0x29b680869f15a>)()
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  command-execute(eval-last-sexp)
```

Unlike https://github.com/minad/consult/issues/1032, there is no option to recompile as the overlay uses a cache.

**Comment by minad** [comment]
Hi, the issue is still a compilation problem. It seems that the byte code of the installed Consult package is outdated. I am not familiar with the way Nix compiles the Emacs packages, but it can only be fixed there.

**Reply by adham-omran** [reply]
I see, I'm working on understanding how the packaging is happening to figure out how to invoke a re-compliation manually. I'll update when I get that completed and try to let the people maintaing the overlay about it.

**Reply by adham-omran** [reply]
The overlay triggered a complete rebuild today. It still reports the same issue. I'll contact the overlay folk today.

![Screenshot from 2024-06-19 14-13-45](https://github.com/minad/consult/assets/24703313/c3b9e04e-7b13-42cc-8542-d4f1d20cb314)

![Screenshot from 2024-06-19 14-13-39](https://github.com/minad/consult/assets/24703313/7721086e-047c-484c-be0c-c4df2703688d)

**Comment by maxc01** [comment, marked as answer]
I also got this issue under Emacs 30. I just deleted consult and reinstalled and the issue is gone.
